presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -397,6 +397,31 @@ public synchronized void updatePartitionStatistics(String databaseName, String t
         return tables;
     }
 
+    @Override
+    public synchronized List<String> getTablesWithParameter(String databaseName, String parameterKey, String parameterValue)
+    {
+        requireNonNull(databaseName, "databaseName is null");
+        requireNonNull(parameterKey, "parameterKey is null");
+        requireNonNull(parameterValue, "parameterValue is null");
+
+        Optional<Database> database = getDatabase(databaseName);
+        if (!database.isPresent()) {
+            return ImmutableList.of();
+        }
+
+        Path databaseMetadataDirectory = getDatabaseMetadataDirectory(databaseName);
+        List<String> tables = getChildSchemaDirectories(databaseMetadataDirectory).stream()
+                .map(Path::getName)
```

Comment:
`getAllViews` is simple because it delegates to `getAllTables` some of the work.
Can you use the same pattern in this method too?

Reply:
@electrum can we change `FileHiveMetastore.getAllViews` like that?

Reply:
Seems fine to me. Unless someone changes the JSON metadata files manually, the Presto view flag won't be set for a non-view table type.